.golangci.yml

```
@@ -44,346 +39,356 @@ linters:
+    # should be removed as the referenced deprecated item is removed from the project.
+      - linters:
+          - staticcheck
+        text: 'SA1019: (bootstrapv1.ClusterStatus|KubeadmConfigSpec.UseExperimentalRetryJoin|scope.Config.Spec.UseExperimentalRetryJoin|DockerMachine.Spec.Bootstrapped|machineStatus.Bootstrapped|dockerMachine.Spec.Backend.Docker.Bootstrapped|dockerMachine.Spec.Bootstrapped|devMachine.Spec.Backend.Docker.Bootstrapped|c.TopologyPlan|clusterv1.ClusterClassVariableMetadata|clusterv1beta1.ClusterClassVariableMetadata|(variable|currentDefinition|specVar|newVariableDefinition|statusVarDefinition).Metadata) is deprecated'
+        # Deprecations forGetIPFamily
+      - linters:
+          - staticcheck
+        text: 'SA1019: cluster.GetIPFamily is deprecated: IPFamily is not a concept in Kubernetes. It was originally introduced in CAPI for CAPD. IPFamily will be dropped in a future release. More details at https://github.com/kubernetes-sigs/cluster-api/issues/7521'
+        # Deprecations for MD revision management
+      - linters:
+          - staticcheck
+        text: 'SA1019: ((deployment|m|md).Spec.RevisionHistoryLimit|clusterv1.RevisionHistoryAnnotation|c.RolloutUndo) is deprecated'
+        # Deprecations for MD revision management
+      - linters:
+          - staticcheck
+        text: 'SA1019: (m|md).Spec.ProgressDeadlineSeconds is deprecated: This field is deprecated and is going to be removed in the next apiVersion. Please see https://github.com/kubernetes-sigs/cluster-api/issues/11470 for more details.'
+        # Deprecations for MHC MaxUnhealthy, UnhealthyRange
+      - linters:
+          - staticcheck
+        text: 'SA1019: (mhc|m)(.Spec.MaxUnhealthy|.Spec.UnhealthyRange) is deprecated'
+        # Specific exclude rules for deprecated packages that are still part of the codebase. These
+        # should be removed as the referenced deprecated packages are removed from the project.
+      - linters:
+          - staticcheck
+        text: 'SA1019: .* is deprecated: This package will be removed in one of the next releases.'
+        # Specific exclude rules for deprecated types that are still part of the codebase. These
+        # should be removed as the referenced deprecated types are removed from the project.
+      - linters:
+          - staticcheck
+        text: 'SA1019: (clusterv1alpha3.*|clusterv1alpha4.*) is deprecated: This type will be removed in one of the next releases.'
+        # Specific exclude rules for deprecated feature flags
+      - linters:
+          - staticcheck
+        text: 'SA1019: feature.ClusterResourceSet is deprecated: ClusterResourceSet feature is now GA and the corresponding feature flag will be removed in 1.12 release.'
+      - linters:
+          - staticcheck
+        text: 'SA1019: .* is deprecated: This field is deprecated and is going to be removed when support for v1beta1 will be dropped.'
```

```suggestion
        # v1Beta1 deprecated fields
      - linters:
          - staticcheck
        text: 'SA1019: .*\.Deprecated\.V1Beta1.* is deprecated'
```

Let's keep the old regex here :-)